foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConversionManager.java

```diff
@@ -1040,6 +1082,48 @@ protected java.time.OffsetTime convertObjectToOffsetTime(Object sourceObject) th
         return offsetTime;
     }
 
+    /**
+     * INTERNAL: Build a valid instance of java.time.Year from the given
+     * source object.
+     *
+     * @param sourceObject
+     *            Valid object of class java.sql.Timestamp, String,
+     *            java.util.Date or Int
+     */
+    protected java.time.Year convertObjectToYear(Object sourceObject) throws ConversionException {
+        java.time.Year year = null;
+
+        if (sourceObject instanceof java.time.Year) {
+            return (java.time.Year) sourceObject;
+        }
+
+        if (sourceObject instanceof String) {
+            try {
+                year = java.time.Year.of(Integer.valueOf((String) sourceObject));
+            } catch (Exception e) {
+                year = java.time.Year.parse(((String) sourceObject).replace(' ', 'T'), Helper.getDefaultDateTimeFormatter());
+                java.time.LocalTime localTime = java.time.LocalTime.parse(((String) sourceObject).replace(' ', 'T'), Helper.getDefaultDateTimeFormatter());
```

Comment:
this looks unused. Is it needed? What was the intent?

Reply:
Fixed in #2081